theano/gpuarray/opt.py

```diff
@@ -697,6 +698,28 @@ def local_gpua_elemwise(op, context_name, inputs, outputs):
         name = 'Gpu' + name
     if len(outputs) > 1:
         return
+    have_cuda = False
+    have_opencl = False
+    if inputs and isinstance(inputs[0].type, GpuArrayType):
```

Comment:
Did you saw cases when inputs is an empty list? I think in theory, it could happen, but I'm not sure we support it. Keep the condition like this, I just wanted to know if you saw this happen or not.